node/src/protocol/triple.rs

```diff
@@ -2,25 +2,40 @@ use super::message::TripleMessage;
 use crate::types::TripleProtocol;
 use crate::util::AffinePointExt;
 use cait_sith::protocol::{Action, Participant};
-use cait_sith::triples::TripleGenerationOutput;
+use cait_sith::triples::{TriplePub, TripleShare};
 use k256::Secp256k1;
-use std::collections::btree_map::Entry;
-use std::collections::BTreeMap;
+use std::collections::hash_map::Entry;
+use std::collections::{HashMap, VecDeque};
 
 /// Unique number used to identify a specific ongoing triple generation protocol.
 /// Without `TripleId` it would be unclear where to route incoming cait-sith triple generation
 /// messages.
 pub type TripleId = u64;
 
+/// A completed triple.
+pub struct Triple {
+    pub id: TripleId,
+    pub share: TripleShare<Secp256k1>,
+    pub public: TriplePub<Secp256k1>,
+}
+
+/// An ongoing triple generator.
+pub struct TripleGenerator {
+    /// Ongoing cait-sith triple generation protocol.
+    pub protocol: TripleProtocol,
+    /// Whether this triple generation was initiated by the current node.
+    pub mine: bool,
+}
+
 /// Abstracts how triples are generated by providing a way to request a new triple that will be
 /// complete some time in the future and a way to take an already generated triple.
 pub struct TripleManager {
     /// Completed unspent triples
-    // TODO: I put these into `BTreeMap` so we can potentially use the ordering for choosing which
-    // triple to use for consensus, but it might not be necessary.
-    triples: BTreeMap<TripleId, TripleGenerationOutput<Secp256k1>>,
+    triples: HashMap<TripleId, Triple>,
     /// Ongoing triple generation protocols
-    generators: BTreeMap<TripleId, TripleProtocol>,
+    generators: HashMap<TripleId, TripleGenerator>,
```

Comment:
so we know we no longer need the guarantees of BTreeMaps now?

Reply:
Yeah turns out ordering is not necessary